doc/arkode/guide/source/Usage/User_callable.rst

```diff
@@ -3470,6 +3470,45 @@ Retrieve a pointer for user data                       :c:func:`ARKodeGetUserDat
    .. versionadded:: 6.1.0
 
 
+.. c:function:: int ARKodeGetNumRhsEvals(void* arkode_mem, int num_rhs, long int* num_rhs_evals)
```

Comment:
While this unifies all the RHS eval getter, the `num_rhs` argument is still closely tied to the type of stepper. Each stepper already "knows" it's number of RHS' and how many values to write to `num_rhs_evals`, so in that sense the argument is not needed. Unfortunately, it is needed for validation but still doesn't guarantee `num_rhs_evals` has sufficient storage. 

I'd like to pitch what I think is a somewhat cleaner option:
```
int ARKodeGetNumRhsEvals(void* arkode_mem, int partition_idx, long int* num_rhs_evals)
```
This writes a single number to `num_rhs_evals` corresponding to the evals for partition `partition_idx`. If `partition_idx<0` that can be the total across all RHS'. This is essentially what I did for counting evolves in operator splitting:

https://github.com/LLNL/sundials/blob/7cd34e8720423a61b0f7c88edcb61d7ef4be075e/doc/arkode/guide/source/Usage/SplittingStep/User_callable.rst?plain=1#L127-L148

Reply:
I think this sounds good and is a bit cleaner. @drreynolds what do you think?

Reply:
I'm good with either approach.